### ISSUE TITLE
docker run --ulimit <type>=<soft>:<hard>

### ISSUE THREAD

**Issue by malikkal** [vmware/vic#5353]
is this supported or applicable in cVM environment? Thanks.

**Comment by anchal-agrawal** [comment]
We currently don't support `--ulimit` - pinging @hickeng and @caglar10ur for the estimate and scope of this issue.

**Comment by caglar10ur** [comment]
I think the real question is why we need them? Docker does it like this cause setting ulimits in a linux container requires privileges, we don't have that problem so IMHO not sure whether we should even try to mimic this.

**Comment by hickeng** [comment]
@caglar10ur You're correct, but I don't think we push up the defaults to unlimited. We should, at a minimum, ensure that ulimit caps are disable by default in the cVM case.

**Comment by malikkal** [comment]
One of the application wont start, if the ulimit for nofile is not set as required. I am thinking of handling this via entrypoint script before starting the application process.

**Comment by hickeng** [comment]
@malikkal As @anchal-agrawal mentioned it's not currently supported, and it's not a priority because of the reasons @caglar10ur detailed.
The current defaults are:
```
/ # ulimit -a
-f: file size (blocks)             unlimited
-t: cpu time (seconds)             unlimited
-d: data seg size (kb)             unlimited
-s: stack size (kb)                8192
-c: core file size (blocks)        0
-m: resident set size (kb)         unlimited
-l: locked memory (kb)             64
-p: processes                      7780
-n: file descriptors               1024
-v: address space (kb)             unlimited
-w: locks                          unlimited
-e: scheduling priority            0
-r: real-time priority             0
```
The primary reason I can see for bumping the priority of this would be if the _defaults_ are problematic and need adjusting. I'd be easily persuaded to have all of them set to unlimited by default, but is there something in specific that you need to be able to adjust on a container by container basis?

**Comment by malikkal** [comment]
@hickeng Thank you.

A bit of background. 1. Developer folks were trying to use  run --ulimit <type>=<soft>:<hard> and I couldn't immediately find, whether this is supported or not. 2. I wasn't sure what are the implications tweaking this in cVM. 3. I am okay to adjust this via entrypoint specific to their requirements, where necessary.

However, it would be great, if I could quickly refer to a doc, where which params of run are supported and which are not and what is planned. This would help us plan how to build the image and tweak things as we progress. The current [doc](https://vmware.github.io/vic-product/assets/files/html/1.1/vic_app_dev/container_operations.html) links to docker documentation, which could give the impression that all params are supported.

**Comment by hickeng** [comment]
@caglar10ur please can you run through the `run` options for @malikkal 
@stuclem I had thought we had this information documented but I've also been unable to find an up-to-date reference - is this something we're failing to find or that we've failed to publish?

**Comment by caglar10ur** [comment]
@malikkal @stuclem @hickeng 

Just went over the create/start/run parameters and following is the list of options VIC support at this time.

```
--attach, -a
--cidfile
--cpuset-cpus
--entrypoint
--env, -e
--env-file
--help
--interactive, -i
--ip
--link
--memory, m
--name
--net
--net-alias
--publish, -p
--rm
--stop-signal
--stop-timeout
--tty, -t
--user, -u
--volume, -v
--workdir, -w
```

Will update this issue once we have a list of options we are planning to support

**Comment by malikkal** [comment]
Thanks @caglar10ur Noted.

**Comment by stuclem** [comment]
@anchal-agrawal @caglar10ur @hickeng, the list of supported options for create/start/run that we document in https://vmware.github.io/vic-product/assets/files/html/1.1/vic_app_dev/container_operations.html#container is nothing like the list that @caglar10ur provided in https://github.com/vmware/vic/issues/5353#issuecomment-307247871. Looks like the doc is very inadequate here.  Created https://github.com/vmware/vic-product/issues/273 to get the docs updated.

@caglar10ur is the list above valid for 1.1? Thanks

**Comment by hickeng** [comment]
There's a very slightly different use-case than initially detailed that bumps the priority of this (see #6279):
* running a container as a non-root user

If running as non-root then the ability to set these values without restriction in the container is contingent on a configured privilege escalation mechanism such as sudoers or a custom suid binary.

**Comment by renmaosheng** [comment]
closing this as #6279 is still opened for tracking.
